storage/src/adb/any/sync/client.rs

```diff
@@ -240,8 +235,177 @@ where
+                upper_bound_pos: new_target.upper_bound_ops,
+            });
+        }
+        if new_target.lower_bound_ops < old_target.lower_bound_ops
+            || new_target.upper_bound_ops < old_target.upper_bound_ops
+        {
+            return Err(Error::SyncTargetMovedBackward {
+                old: Box::new(old_target.clone()),
+                new: Box::new(new_target.clone()),
+            });
+        }
+        if new_target.hash == old_target.hash {
+            return Err(Error::SyncTargetHashUnchanged);
+        }
+        Ok(())
+    }
+
+    /// Update the sync target to the most recent target update, if any.
+    async fn handle_target_updates(mut self) -> Result<Self, Error> {
+        let update_receiver = match &mut self {
+            Client::FetchData { config, .. } => &mut config.update_receiver,
+            Client::ApplyData { config, .. } => &mut config.update_receiver,
+            Client::Done { .. } => {
+                warn!("Ignoring target update - sync already completed");
+                return Ok(self);
+            }
+        };
+        let mut new_target = None;
+        if let Some(ref mut receiver) = update_receiver {
+            // Only apply the last update, if any.
+            while let Ok(Some(new_target_)) = receiver.try_next() {
+                new_target = Some(new_target_);
+            }
+        }
+
+        let Some(new_target) = new_target else {
+            // There is no new target to apply.
+            return Ok(self);
+        };
+
+        let (mut config, mut log, metrics) = match self {
+            Client::FetchData {
+                config,
+                log,
+                metrics,
+                ..
+            } => (config, log, metrics),
+            Client::ApplyData {
+                config,
+                log,
+                metrics,
+                ..
+            } => (config, log, metrics),
+            Client::Done { .. } => {
+                return Ok(self);
+            }
+        };
+        Self::validate_target_update(&config.target, &new_target)?;
+
+        info!(
+            old_target = ?config.target,
+            new_target = ?new_target,
+            "Applying target update"
+        );
+
+        // Check if the existing log contains data in the updated sync range
+        let log_size = log
+            .size()
+            .await
+            .map_err(|e| Error::Adb(adb::Error::JournalError(e)))?;
+
+        let log = if log_size <= new_target.lower_bound_ops {
+            // Log is stale (last element is before the new lower bound)
+            // Reinitialize the log with the new sync range
+            log.close()
+                .await
+                .map_err(|e| Error::Adb(adb::Error::JournalError(e)))?;
+
+            debug!(
+                log_size,
+                old_bounds = ?config.target,
+                new_bounds = ?new_target,
+                "Log is stale, reinitializing"
```

Comment:
super nitty question but do we want to standardize on capitalizing the first word of any log line? I think I've been leaving it all lowercase unless it's really a full sentence with periods.

Reply:
Yup, you're right. https://github.com/commonwarexyz/monorepo/pull/1295 changes the logs in `adb` to start with lowercase.